test/integration/test_celery_user_rate_limit.py

```diff
@@ -0,0 +1,169 @@
+    return task_user_id
+
+
+@lru_cache()
+def sqlite_url():
+    path = tempfile.NamedTemporaryFile().name
+    dburl = f"sqlite:///{path}"
+    init_database(dburl)
+    return dburl
+
+
+@lru_cache()
+def setup_users(dburl: str, num_users: int = 2):
+    """
+    Setup test users in galaxy_user table with user id's starting from 2 because
+    we assume there always exists a user with id = 1.
+    This is because the new celery_user_rate_limit table has
+    a user_id with a foreign key pointing to galaxy_user table.
+    """
+    expected_user_ids = [i for i in range(2, num_users + 1)]
+    with sqlalchemy_engine(dburl) as engine:
+        with engine.begin() as conn:
+            found_user_ids = conn.scalars(
+                text("select id from galaxy_user where id between 1 and :high"), {"high": num_users}
+            ).all()
+            if len(expected_user_ids) > len(found_user_ids):
+                user_ids_to_add = set(expected_user_ids).difference(found_user_ids)
+                for user_id in user_ids_to_add:
+                    conn.execute(
+                        text("insert into galaxy_user(id, active, email, password) values (:id, :active, :email, :pw)"),
+                        [{"id": user_id, "active": True, "email": "e", "pw": "p"}],
+                    )
+
+
+class TestCeleryUserRateLimitIntegration(IntegrationTestCase):
+    """
+    Test feature that limits the number of celery task executions
+    per user per second. This is implemented by the celery_user_rate_limit
+    config parameter whose default value is 0.0 meaning no rate limit.
+    For each of a set of users it submits runs a task num_calls times.
+    For the test to succeed the total duration of completion of all task
+    executions should be equal to the tasks_per_user_per_sec times
+    num_calls-1. This is because tasks for different users should be
+    executed in parallel. In addition, we verify that the total
+    duration for execution of tasks for each user equals the total
+    task duration, again because tasks for different users should
+    run in parallel.
+    """
+
+    def setUp(self):
+        super().setUp()
+
+    def _test_mock_pass_user_id_task(self, users: Iterable[int], num_calls: int, tasks_per_user_per_sec: float):
+        expected_duration: float
+        if tasks_per_user_per_sec == 0.0:
+            expected_duration = 0.0
+            expected_duration_lbound = 0.0
+        else:
+            secs_between_tasks_per_user = 1 / tasks_per_user_per_sec
+            expected_duration = secs_between_tasks_per_user * (num_calls - 1)
+            expected_duration_lbound = expected_duration - 4
+        expected_duration_hbound = expected_duration + 4
+        start_time = datetime.datetime.utcnow()
+        timer = ExecutionTimer()
+        results: Dict[int, List[AsyncResult]] = {}
+        for user in users:
+            user_results: List[AsyncResult] = []
+            for _i in range(num_calls):  # type: ignore
```

Comment:
```suggestion
            for _ in range(num_calls):  # type: ignore
```
Just use `_` for a throwaway counter variable (pythonic idiom)

Reply:
fine

Reply:
same as the comment above: we try to keep the degree of "surprise" to a minimum. It's immediately clear what's happening when the iter variable is an underscore. It's not because it's pythonic - it's for convenience.